modules/plugin/http-commons/src/main/java/org/geotools/http/commons/MultithreadedHttpClient.java

```diff
@@ -250,59 +222,68 @@ public void setPassword(String password) {
     }
 
     private void resetCredentials() {
-        client.getState().clearCredentials();
         if (user != null && password != null) {
             AuthScope authscope = AuthScope.ANY;
             Credentials credentials = new UsernamePasswordCredentials(user, password);
-
-            client.getParams().setAuthenticationPreemptive(true);
-            client.getState().setCredentials(authscope, credentials);
-        } else {
-            client.getParams().setAuthenticationPreemptive(false);
+            // TODO - check if this works for all types of auth or do we need to look it up?
```

Comment:
I see another bit of code above where credentials are base64 encoded and set in the headers... so is this not working as it should?

Reply:
It seems to depend on if you are using GET or POST and I'm not entirely sure what setting the credentials here does or doesn't affect. Also not sure if this code can handle anything but BASIC auth (as that's all that is tested) or if in fact we support anything else

Reply:
I've managed to fix this up - it seems the problem was the wiremock doesn't ask for basic authentication so the client doesn't send it!